src/stateless/slider.vue

```diff
@@ -0,0 +1,545 @@
+        this.labelPadding = 8
+    },
+    mounted () {
+        this.isDomReady = true
+
+        // threshold for ticks disappearing under labels
+        this.minThreshold = this.$refs.min.clientWidth + this.labelPadding
+        this.maxThreshold = this.sliderWidth - this.$refs.max.clientWidth - this.labelPadding
+    },
+    methods: {
+        startDrag (e) {
+            if (this.disabled) return
+
+            this.isChanged = true
+            this.isDragging = true
+
+            this.setPosition(e)
+            window.addEventListener('mouseup', this.stopDrag)
+            window.addEventListener('mousemove', this.setPosition)
+            this.$refs.bar.focus()
+        },
+        setPosition (e) {
+            let percent = (e.clientX - this.sliderOffset) / this.sliderWidth
+            percent = Math.min(Math.max(0, percent), 0.999999)
+            let edgeStepOffset = this.stepPercentage / 2
+
+            let value = (Math.floor((percent - edgeStepOffset) / this.stepPercentage) + 1) * this.step + this.min
+            this.$emit('input', this.roundValue(value))
+
+            e.preventDefault()
+        },
+        stopDrag () {
+            this.isDragging = false
+            window.removeEventListener('mouseup', this.stopDrag)
+            window.removeEventListener('mousemove', this.setPosition)
+        },
+        onKeyboardInput (e) {
+            if (this.disabled) return
+            this.isChanged = true
+
+            let key = e.keyCode
+
+            if ((key === 38 || key === 39) && this.index < this.stepsCount)
+                this.$emit('input', this.value + this.step)
+            else if ((key === 37 || key === 40) && this.index > 0)
+                this.$emit('input', this.value - this.step)
+
+            e.preventDefault()
+        },
+        roundValue (value) {
+            let roundingFactor = 1
+            let decimalsPart = this.step.toString().split('.')[1]
+            if (decimalsPart) {
+                roundingFactor = parseInt('1' + '0'.repeat(decimalsPart.length), 10)
+            }
+            return Math.round(value * roundingFactor) / roundingFactor
+        },
+        handleInput (value) {
+            let number = parseFloat(value)
+            this.$emit('input', isNaN(number) ? null : number)
+        },
+        isUnitActive (index) {
+            return index / (this.ticksCount + 1) <= this.percentage
+        },
+        isValidInput (value, onError) {
+            return (value < this.min || this.max < value) ? '' : null
+        },
+        tickClass (index) {
+            let hidden = false
+            if (this.isDomReady) {
+                let position = index / (this.ticksCount + 1) * this.sliderWidth
+                hidden = position <= this.minThreshold || position >= this.maxThreshold
+            }
+
+            return {
+                'active': this.isUnitActive(index),
+                'hidden': hidden,
+            }
+        },
+        labelClass (index) {
```

Comment:
This could be a computed property `labelsClass` that returns an object like `{ min: { active: true }, max: { active: false } }`